tests/integration/pilot/tunneling/tunneling_test.go

```diff
@@ -0,0 +1,416 @@
+		// Flag --insecure must be passed, because self-signed certificates are used.
+		"curl --insecure -s -o /dev/null -w '%%{http_code}' %s://external-app.testdomain:%d/test/%s",
+		protocol, selectPort(protocol), testName)
+
+	stdout, _, err := kubeClient.PodExec(sleepPodName, meshNs, "sleep", testRequestCmd)
+	if err != nil {
+		return fmt.Errorf("failed to execute command in %s pod: %v: %s", sleepPodName, err, stdout)
+	}
+	if stdout != "'200'" {
+		return fmt.Errorf("test request failed: %s", stdout)
+	}
+	return nil
+}
+
+func verifyThatRequestWasTunneled(ctx framework.TestContext, externalNs, expectedSourceIP, expectedProtocol, expectedPath string) error {
+	kubeClient := ctx.Clusters().Default()
+
+	externalAppPodName := getPodName(ctx, externalNs, "external-app")
+	getAccessLogCmd := "cat /var/log/nginx/access.log"
+	stdout, _, err := kubeClient.PodExec(externalAppPodName, externalNs, "external-app", getAccessLogCmd)
+	if err != nil {
+		return fmt.Errorf("failed to get logs of external-app: %v", err)
+	}
+
+	expectedLog := fmt.Sprintf("%s - GET %s://external-app.testdomain:%d/test/%s - 200",
+		expectedSourceIP, expectedProtocol, selectPort(expectedProtocol), expectedPath)
+	if !strings.Contains(stdout, expectedLog) {
+		return fmt.Errorf("failed to find expected log: %s; logs: %s", expectedLog, stdout)
+	}
+	return nil
+}
+
+// makeExternalServicesResolvable sets host aliases for external-app and external-forward-proxy in sleep
+// and istio-egressgateway deployments. Without host aliases domains of external-app and external-forward-proxy
+// would not be resolvable, because there are no Kubernetes services for them. Instead, Istio service entries are used.
+// The reason for this approach is to simulate communication with mesh-external services.
+func makeExternalServicesResolvable(ctx framework.TestContext, externalNs, meshNs string) {
+	externalAppIP := getPodIP(ctx, externalNs, "external-app")
+	// "127.0.0.1" is used as the IP address for external-forward-proxy.testdomain,
+	// because external-forward-proxy will not need to resolve its own domain name
+	// and setting its current IP address doesn't make sense, because it will change after this update
+	updateHostAliasesInDeploymentOrFail(ctx, externalNs, "external-forward-proxy", externalAppIP, "127.0.0.1")
+	externalForwardProxyIP := getPodIP(ctx, externalNs, "external-forward-proxy")
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		updateHostAliasesInDeploymentOrFail(ctx, "istio-system", "istio-egressgateway", externalAppIP, externalForwardProxyIP)
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		updateHostAliasesInDeploymentOrFail(ctx, meshNs, "sleep", externalAppIP, externalForwardProxyIP)
+	}()
+
+	wg.Wait()
+}
+
+func updateHostAliasesInDeploymentOrFail(ctx framework.TestContext, ns, name, externalAppIP, externalForwardProxyIP string) {
+	kubeClient := ctx.Clusters().Default().Kube()
+
+	scaleDeploymentOrFail(ctx, ns, name, 0)
```

Comment:
why scaling down?

Reply:
It's no longer relevant, because I removed this function.